packages/nodes-base/nodes/Clockify/ClockifyTrigger.node.ts

```diff
@@ -109,16 +109,13 @@ export class ClockifyTrigger implements INodeType {
 				qs.start = webhookData.lastTimeChecked;
 				qs.end = moment().tz(workflowTimezone).format('YYYY-MM-DDTHH:mm:ss') + 'Z';
 				qs.hydrated = true;
-				qs['in-progress'] = false;
+				qs[ 'in-progress' ] = false;
 				break;
 		}
 
 		result = await clockifyApiRequest.call(this, 'GET', resource, {}, qs);
 		webhookData.lastTimeChecked = qs.end;
 
-		if (Array.isArray(result) && result.length !== 0) {
-			result = [this.helpers.returnJsonArray(result)];
-		}
-		return result;
+		return [ this.helpers.returnJsonArray(result) ];
```

Comment:
Thanks for the investigation.

This fixes the problem but still does not have the desired effect.

The way it is working is that we will always have an execution even if there is no data.

I was checking the way `poll` triggers work and we should return `null` if there is no data.

So I think it's best if we add a check like this:

```
if (result.length === 0) {
    return null;
}
```

This way the workflow will only trigger executions when there is data to run.

Reply:
Hey @krynble , I missed that side effect, fixed now!

Reply:
Thanks a lot :)